### PR TITLE
Login with a registered FIDO2 device

### PR DESCRIPTION
**Pull request by roxanemace**
**Asana task**: https://app.asana.com/0/1141396442496489/1181397055064771

### CODE REVIEW

File: sdk-core/src/main/java/com/reach5/identity/sdk/core/ReachFive.kt [hunk 1]
```diff
@@ -501,13 +505,51 @@ class ReachFive (
...
+                success = {
+                    val fido2ApiClient = Fido.getFido2ApiClient(activity)
+                    val fido2PendingIntentTask = fido2ApiClient.getSignPendingIntent(it.toFido2Model())
+                    fido2PendingIntentTask.addOnSuccessListener { fido2PendingIntent ->
+                        if (fido2PendingIntent != null) {
+                            try {
+                                Log.d(TAG, "Launching Fido2 Pending Intent")
+                                activity.startIntentSenderForResult(fido2PendingIntent.intentSender, loginRequestCode, null, 0, 0, 0)
+                            } catch (e: IntentSender.SendIntentException) {
+                                e.printStackTrace()
```
Comment (FabienPennequin) [inline sdk-core/src/main/java/com/reach5/identity/sdk/core/ReachFive.kt:525]: intended?

File: sdk-core/src/main/java/com/reach5/identity/sdk/core/models/requests/webAuthn/AuthenticationPublicKeyCredential.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import com.google.gson.annotations.SerializedName
+import com.reach5.identity.sdk.core.utils.WebAuthn.encodeToBase64
+import com.reach5.identity.sdk.core.utils.WebAuthn.publicKeyCredentialType
+import kotlinx.android.parcel.Parcelize
+
+object WebAuthnAuthentication{
+    fun createAuthenticationPublicKeyCredential(authenticatorAssertionResponse: AuthenticatorAssertionResponse): AuthenticationPublicKeyCredential {
+        return AuthenticationPublicKeyCredential(
+            id = encodeToBase64(authenticatorAssertionResponse.keyHandle),
+            rawId = encodeToBase64(authenticatorAssertionResponse.keyHandle),
```
Comment (FabienPennequin) [inline sdk-core/src/main/java/com/reach5/identity/sdk/core/models/requests/webAuthn/AuthenticationPublicKeyCredential.kt:14]: id and rawId are equals? It's intended?

File: sdk-core/src/main/java/com/reach5/identity/sdk/core/ReachFive.kt (hunk 1 above)
Comment (roxanemace) [reply sdk-core/src/main/java/com/reach5/identity/sdk/core/ReachFive.kt:525]: Sorry, what do you mean? Indentation?

File: sdk-core/src/main/java/com/reach5/identity/sdk/core/models/requests/webAuthn/AuthenticationPublicKeyCredential.kt (hunk 2 above)
Comment (roxanemace) [reply sdk-core/src/main/java/com/reach5/identity/sdk/core/models/requests/webAuthn/AuthenticationPublicKeyCredential.kt:14]: `raw_id` is the binary form of the `id`, but we need to keep both.

Do you confirm @rchapel ?

Comment (rchapel) [reply sdk-core/src/main/java/com/reach5/identity/sdk/core/models/requests/webAuthn/AuthenticationPublicKeyCredential.kt:14]: Yes they are the same. We are keeping both to keep messages as close as possible from the specification.
Actually `rawId` is a byte buffer and `id` the Base64 encoding from this byte buffer.

Comment (roxanemace) [reply sdk-core/src/main/java/com/reach5/identity/sdk/core/models/requests/webAuthn/AuthenticationPublicKeyCredential.kt:14]: So @rchapel is it ok if I encode in Base64 the `rawId` too?

Comment (rchapel) [reply sdk-core/src/main/java/com/reach5/identity/sdk/core/models/requests/webAuthn/AuthenticationPublicKeyCredential.kt:14]: Yes if it is needed. But be careful to not encode twice the id in base 64 (if it is already base64 encoded).

Comment (roxanemace) [reply sdk-core/src/main/java/com/reach5/identity/sdk/core/models/requests/webAuthn/AuthenticationPublicKeyCredential.kt:14]: No worries, they are not. If I don't encode them I'll have the error `bad base64 encoding` from the server.

File: sdk-core/src/main/java/com/reach5/identity/sdk/core/ReachFive.kt (hunk 1 above)
Comment (FabienPennequin) [reply sdk-core/src/main/java/com/reach5/identity/sdk/core/ReachFive.kt:525]: We need to call the `printStackTrace` function here?

**Review by StanWarz — APPROVED** [review]

**Pull request merged**